File: 01/01.py
from timing_util import print_elapsed, timestamp_nano


def get_data(content: str) -> list[list[int]]:
    return [list(map(int, elf.split('\n'))) for elf in content.split('\n\n')]


def part12(data: list[list[int]]) -> tuple[int, int]:
    res = sorted([sum(elf) for elf in data], reverse=True)[:3]
    return res[0], sum(res)


if __name__ == '__main__':
    start = timestamp_nano()

    with open('01/input.txt') as in_file:
        data = get_data(in_file.read())

    p1, p2 = part12(data)
    print(f'part1: {p1}')
    print(f'part2: {p2}')

    print_elapsed(start)

File: 01/01.jl
timestamp_nano() = Int(round(time() * 1e9))

get_data() = open("01/input.txt") do in_file
    content = read(in_file, String)
    sort(
        collect(
            sum(
                parse.(Int, eachsplit(elf, "\n"))
            ) for elf in eachsplit(content, "\n\n")),
        rev=true
    )[1:3]
end

start = timestamp_nano()

data = get_data()

println("part1: ", data[1])
println("part2: ", sum(data))

end_ = timestamp_nano()
println("time: ", round((end_ - start) / 1e6, digits=3), "ms")
